src/test/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/EventGen.scala

```diff
@@ -0,0 +1,351 @@
+
+  val ipAddress: Gen[String] =
+    Gen.oneOf(ipv4Address, ipv6Address)
+
+  val platform: Gen[String] = Gen.oneOf("web", "mob", "app")
+
+  val eventType: Gen[String] = Gen.oneOf("page_view", "page_ping", "transaction", "unstruct")
+
+  val contexts: Contexts = parse(EventSpec.contextsJson)
+    .flatMap(_.as[Contexts])
+    .getOrElse(throw new UnsupportedOperationException("can't decode contexts"))
+
+  val unstruct: UnstructEvent = parse(EventSpec.unstructJson)
+    .flatMap(_.as[UnstructEvent])
+    .getOrElse(throw new UnsupportedOperationException("can't decode unstructured event"))
+
+  val derived_contexts: Contexts = parse(EventSpec.derivedContextsJson)
+    .flatMap(_.as[Contexts])
+    .getOrElse(throw new UnsupportedOperationException("can't decode derived contexts"))
+
+  val event: Gen[Event] =
+    for {
+      app_id <- Gen.option(strGen(512, Gen.alphaNumChar))
+      platform <- Gen.option(platform)
+      etl_tstamp <- Gen.option(instantGen)
+      collector_tstamp <- instantGen
+      dvce_created_tstamp <- Gen.option(instantGen)
+      event <- Gen.option(eventType)
+      event_id <- Gen.uuid
+      txn_id <- Gen.option(Gen.chooseNum(1, 10000))
+      name_tracker <- Gen.option(strGen(256, Gen.alphaNumChar))
+      v_tracker <- Gen.option(strGen(256, Gen.alphaNumChar))
+      v_collector <- strGen(512, Gen.alphaNumChar)
+      v_etl <- strGen(512, Gen.alphaNumChar)
+      user_id <- Gen.option(Gen.uuid).map(_.map(_.toString()))
+      user_ipaddress <- Gen.option(ipAddress)
+      user_fingerprint <- Gen.option(strGen(512, Gen.alphaNumChar))
+      domain_userid <- Gen.option(Gen.uuid).map(_.map(_.toString()))
+      domain_sessionidx <- Gen.option(Gen.chooseNum(1, 10000))
+      network_userid <- Gen.option(Gen.uuid).map(_.map(_.toString()))
+      geo_country <- Gen.option(strGen(3, Gen.alphaUpperChar))
+      geo_region <- Gen.option(strGen(100, Gen.alphaNumChar))
+      geo_city <- Gen.option(strGen(512, Gen.alphaChar))
+      geo_zipcode <- Gen.option(strGen(6, Gen.alphaNumChar))
+      geo_latitude <- Gen.option(Arbitrary.arbitrary[Double])
+      geo_longitude  <- Gen.option(Arbitrary.arbitrary[Double])
+      geo_region_name <- Gen.option(strGen(512, Gen.alphaChar))
+      ip_isp <- Gen.option(strGen(512, Gen.alphaNumChar))
+      ip_organization <- Gen.option(strGen(512, Gen.alphaNumChar))
+      ip_domain <- Gen.option(strGen(512, Gen.alphaNumChar))
+      ip_netspeed <- Gen.option(strGen(50, Gen.alphaNumChar))
+      page_url <- Gen.option(strGen(512, Gen.alphaNumChar))
+      page_title <- Gen.option(strGen(512, Gen.alphaNumChar))
+      page_referrer <- Gen.option(strGen(512, Gen.alphaNumChar))
+      page_urlscheme <- Gen.option(strGen(10, Gen.alphaNumChar))
+      page_urlhost <- Gen.option(strGen(512, Gen.alphaNumChar))
+      page_urlport <- Gen.option(Gen.chooseNum(1, 65000))
+      page_urlpath <- Gen.option(strGen(512, Gen.alphaNumChar))
+      page_urlquery <- Gen.option(strGen(512, Gen.alphaNumChar))
```

Comment:
Technically, this should be key-value pairs, delimited by `&`, not just a random string.